src/ucp/core/ucp_ep.c

```diff
@@ -3472,8 +3542,15 @@ ucs_status_t ucp_ep_query(ucp_ep_h ep, ucp_ep_attr_t *attr)
 
     if (attr->field_mask &
         (UCP_EP_ATTR_FIELD_LOCAL_SOCKADDR | UCP_EP_ATTR_FIELD_REMOTE_SOCKADDR)) {
-        return ucp_ep_query_sockaddr(ep, attr);
+        status = ucp_ep_query_sockaddr(ep, attr);
     }
 
-    return UCS_OK;
+    if (status != UCS_OK) {
+        return status;
+    }
```

Comment:
pls move this code right after line 3545. also, pls add a similar block right after line 3553 , replace "return status" in line 3555 by "return UCS_OK", and remove the initialization of status to UCS_OK in line 3533. this aligns with other error handling flows in the code.